File: movies/src/main/java/com/puntored/movies/entity/Inventory.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.puntored.movies.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

/**
 *
 * @author heiderarellano
 */
@Entity
@Data
@Schema(name = "Inventory", description = "Inventario de Películas asociado a tiendas")
@Table(name = "inventory")
public class Inventory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Schema(description = "identificador único de inventario")
    @Column(name = "inventory_id")
    private long inventoryId;

    @ManyToOne
    @JoinColumn(name = "film_id")
    private Film film;

    @ManyToOne
    @JoinColumn(name = "store_id")
    private Store store;

    @Schema(description = "Cantidad de películas disponible para renta")
    private int quantity;
}

File: movies-python/models/inventory.py
from sqlalchemy import Table, Column, ForeignKey
from sqlalchemy.sql.sqltypes import String, Integer
from config.database import meta, engine

inventory = Table(
    "inventory",
    meta,
    Column("inventory_id", Integer, primary_key=True),
    Column("film_id", Integer, ForeignKey("film.film_id")),
    Column("store_id", Integer, ForeignKey("store.store_id")),
    Column("quantity", Integer, nullable=False)
)

meta.create_all(engine)